Application/EdFi.Ods.Common/Infrastructure/Interceptors/EdFiOdsInterceptor.cs

```diff
@@ -0,0 +1,69 @@
+﻿// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System;
+using System.Text.RegularExpressions;
+using EdFi.Ods.Common.Conventions;
+using EdFi.Ods.Common.Security.CustomViewBased;
+using NHibernate;
+using NHibernate.SqlCommand;
+
+namespace EdFi.Ods.Common.Infrastructure.Interceptors
+{
+    public class EdFiOdsInterceptor : EmptyInterceptor
+    {
+        private const string SpaceLiteral = " ";
+
+        private static readonly Regex regex = new(
+            @$"(?<join>inner join\s+)\(?(?<basisEntity>\w+\.\w+) (?<alias>(?<aliasPrefix>{CustomViewHelpers.CustomViewAliasPrefixBase}[a-f\d]{{4}}_).*?\d_).*?(?<onClause>on\s+)",
+            RegexOptions.Compiled);
+
+        public override bool? IsTransient(object entity)
+        {
+            var property = entity.GetType()
+                                 .GetProperty("CreateDate");
+
+            if (property != null)
+            {
+                bool isTransient = Convert.ToDateTime(property.GetValue(entity)) == default(DateTime);
+
+                return isTransient;
+            }
+
+            return base.IsTransient(entity);
+        }
+
+        public override SqlString OnPrepareStatement(SqlString sql)
+        {
+            // Determine if we need to perform custom authorization view SQL manipulation
+            if (sql.IndexOf(CustomViewHelpers.CustomViewAliasPrefixBase, 0, sql.Length, StringComparison.Ordinal) >= 0)
```

Comment:
Replaces the basis entity with the custom authorization view in the generated SQL, just prior to execution.